### PR TITLE
issue-2387: remove endpoint if volume is not found

### PR DESCRIPTION
**Pull request by antonmyagkov**
issue: #2387

### CODE REVIEW

File: cloud/blockstore/tests/e2e-tests/test.py [hunk 1]
```diff
@@ -227,3 +229,77 @@ def test_resize_device(with_netlink, with_endpoint_proxy):
...
+    except subprocess.CalledProcessError as e:
+        log_called_process_error(e)
+        raise
+
+    cleanup_after_test(env)
+
+    shutil.rmtree(endpoints_dir)
+    shutil.copytree(backup_endpoints_dir, endpoints_dir)
+    env, run = init()
+    assert not os.listdir(endpoints_dir)
```
Comment (jkuradobery) [inline cloud/blockstore/tests/e2e-tests/test.py:304]: Are you sure that we do not need to wait for endpoint to be deleted?

File: cloud/blockstore/tests/e2e-tests/test.py [hunk 2]
```diff
@@ -227,3 +229,77 @@ def test_resize_device(with_netlink, with_endpoint_proxy):
...
+
+        result = run(
+            "stopendpoint",
+            "--socket",
+            socket_path,
+        )
+        assert result.returncode == 0
+
+        result = run(
+            *("destroyvolume", "--disk-id", volume_name),
```
Comment (jkuradobery) [inline cloud/blockstore/tests/e2e-tests/test.py:290]: `*()` is not needed. `**{}` is also not needed.

File: cloud/blockstore/tests/e2e-tests/test.py [hunk 3]
```diff
@@ -227,3 +229,77 @@ def test_resize_device(with_netlink, with_endpoint_proxy):
...
+    # 2. create volume and start endpoint
+    # 4. copy endpoint to backup folder
+    # 5. stop endpoint and remove volume
+    # 6. stop nbs
+    # 7. copy endpoint from backup folder to endpoints folder
+    # 8. run nbs
+    # 9. check that endpoint was removed from endpoints folder
+    env, run = init()
+
+    test_hash = hash(common.context.test_name)
```
[the remark this answers is not in the document]
Comment (antonmyagkov) [reply cloud/blockstore/tests/e2e-tests/test.py:246]: Test name can contain "/" when test is parameterized with path. This is the reason to use hash for endpoint directory in init() function: https://github.com/ydb-platform/nbs/blob/main/cloud/blockstore/tests/e2e-tests/test.py#L40

In the test I just copied this approach to get access to endpoints.

File: cloud/blockstore/tests/e2e-tests/test.py (hunk 1 above)
Comment (antonmyagkov) [reply cloud/blockstore/tests/e2e-tests/test.py:310]: I added list endpoints call to be sure that nbs is running. Endpoint should be deleted before that.

File: cloud/blockstore/tests/e2e-tests/test.py (hunk 2 above)
Comment (antonmyagkov) [reply cloud/blockstore/tests/e2e-tests/test.py]: Fixed

File: cloud/blockstore/tests/e2e-tests/test.py (hunk 1 above)
Comment (antonmyagkov) [reply cloud/blockstore/tests/e2e-tests/test.py:310]: I changed "list endpoints" call to "ping"

Comment (antonmyagkov) [reply cloud/blockstore/tests/e2e-tests/test.py]: Returned back to list endpoints command with --wait-for-restoring argument.

**Review by jkuradobery — APPROVED** [review]

**Review by jkuradobery — APPROVED** [review]

**Review by yegorskii — APPROVED** [review]

**Pull request merged**